client/gutenberg/extensions/giphy/edit.js

```diff
@@ -0,0 +1,193 @@
+		}
+		if ( giphyID ) {
+			return this.fetch( this.urlForId( giphyID ) );
+		}
+
+		return this.fetch( this.urlForSearch( searchText ) );
+	}, 250 );
+
+	urlForSearch = searchText => {
+		const { apiKey } = settings;
+		return `//api.giphy.com/v1/gifs/search?q=${ encodeURIComponent(
+			searchText
+		) }&api_key=${ apiKey }&limit=1`;
+	};
+
+	urlForId = giphyId => {
+		const { apiKey } = settings;
+		return `//api.giphy.com/v1/gifs/${ giphyId }?api_key=${ apiKey }`;
+	};
+
+	splitAndLast = ( array, delimiter ) => {
+		const split = array.split( delimiter );
+		return split[ split.length - 1 ];
+	};
+
+	fetch = url => {
+		const { setAttributes } = this.props;
+		const xhr = new XMLHttpRequest();
+		xhr.open( 'GET', url );
+		xhr.onload = () => {
+			if ( xhr.status === 200 ) {
+				const res = JSON.parse( xhr.responseText );
+				const giphyData = res.data.length > 0 ? res.data[ 0 ] : res.data;
+				// No results
+				if ( ! giphyData.images ) {
+					return;
+				}
+				const topPadding = Math.floor(
+					( giphyData.images.original.height / giphyData.images.original.width ) * 100
+				);
+				const giphyUrl = giphyData.embed_url;
+				setAttributes( { giphyUrl, topPadding } );
+				this.maintainFocus( 500 );
+			} else {
+				// Error handling TK
+			}
+		};
+		xhr.send();
+	};
+
+	setFocus = () => {
+		this.maintainFocus();
+		this.textControlRef.current.querySelector( 'input' ).focus();
+	};
+
+	maintainFocus = ( timeoutDuration = 3500 ) => {
+		this.setState( { focus: true }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+			if ( this.hasSearchText() ) {
+				this.timer = setTimeout( () => {
+					this.setState( { focus: false } );
+				}, timeoutDuration );
+			}
+		} );
+	};
+
+	clearFocus = () => {
+		this.setState( { focus: false }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+		} );
+	};
+
+	hasSearchText = () => {
+		const { attributes } = this.props;
+		const { searchText } = attributes;
+		return searchText && searchText.length > 0;
+	};
+
+	render() {
+		const { attributes, className, isSelected, setAttributes } = this.props;
+		const { align, caption, giphyUrl, searchText, topPadding } = attributes;
+		const { focus } = this.state;
+		const style = {
+			paddingTop: `${ topPadding }%`,
+		};
+		const classes = classNames( className, `align${ align }` );
+		const textControlClasses = classNames(
+			'wp-block-jetpack-giphy_text-input-field',
+			focus || ! this.hasSearchText() ? 'has-focus' : 'no-focus'
+		);
+		return (
+			<Fragment>
+				<BlockControls>
+					<BlockAlignmentToolbar
+						onChange={ value => setAttributes( { align: value } ) }
+						value={ align }
+					/>
+				</BlockControls>
+				<div className={ classes }>
+					<figure style={ style }>
+						<div
+							className="wp-block-jetpack-giphy_cover"
+							onClick={ this.setFocus }
+							onKeyDown={ this.setFocus }
+							ref={ this.textControlRef }
+							role="button"
+							tabIndex="0"
```

Comment:
This wasn't working, it was `tabindex` rather than `tabIndex` as React expected.

However, I'm not sure that this should be tab-focusable. We can tab to the input itself, which seems preferable.

Should we drop this?

Reply:
The `tabIndex` was added in response to an ESLint error:

`Elements with the 'button' interactive role must be tabbable  jsx-a11y/interactive-supports-focus`